RHEcosystemAppEng/dbaas-operator
# [DBAAS-982]Remove the installation of observability operator, create ServiceMonitor CR, use obo-prometheus operator

## Description

This PR  uses pre-installed Observability Operator from the OSD environment and removing the installation of observability from DBaaS. The code checks if the observability operator has a subscription, and then creates the related custom resource for observability. 
As shown in the screenshots, two CRs will be created: one for the ServiceMonitor and another for the MonitoringStack. 

![Screenshot 2023-01-19 at 5 46 00 PM](https://user-images.githubusercontent.com/41960152/213579808-a0063ffa-ec3d-490a-84ec-9bea8583eded.png)
The Observability Team is using their own version of the Prometheus operator (https://github.com/rhobs/obo-prometheus), so the ServiceMonitor CR is created as per the new obo-Prometheus. 
![Screenshot 2023-01-19 at 5 46 46 PM](https://user-images.githubusercontent.com/41960152/213580214-b04e6bbe-133d-4779-b3f4-025e0bc716c5.png)
The second CR for MonitoringStack is created with settings for remote write pushing to observatorium. 
  



![Screenshot 2023-01-19 at 5 47 32 PM](https://user-images.githubusercontent.com/41960152/213580499-f49d8e49-d5ee-4cd6-a650-dbeb464e8e5c.png)

also removing the old subscription of observability operator if installed perviously by RHODA operator. 

see more details about issue here: https://issues.redhat.com/browse/DBAAS-982

## Verification Steps


1.  Deploy the DBaaS operator with this PR changes and verify the required custom resource for observability operator is creating successfully. 
2. Verify Metrics are scraping successfully 
3. Verify remote write is working successfully and metrics is showing on grafana.

## Type of change


## Checklist (Definition of Done)
